contract_sale_invoicing/models/contract.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2018 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    invoicing_sales = fields.Boolean(
+        string='Invoicing Sales',
+        help='If checked to invoce sales will be invoiced on invoice '
+             'recurring invoice creation'
+    )
+
+    @api.multi
+    def _create_invoice(self, invoice=False):
+        if not self.invoicing_sales:
+            return super(AccountAnalyticAccount, self)._create_invoice()
+        sales = self.env['sale.order'].search([
+            ('partner_invoice_id', '=', self.partner_id.id),
+            ('invoice_status', '=', 'to invoice'),
+            ('date_order', '<=',
+             '{} 23:59:59'.format(self.recurring_next_date)),
+        ])
+        invoice_ids = sales.action_invoice_create()
+        invoice = self.env['account.invoice'].browse(invoice_ids)[:1] or False
```

Comment:
What happens if there are more than one invoice?

Reply:
Contract lines will be added to the first invoice created

Reply:
Yeah, of course, but it's a rhetorical question. What I mean is if we should handle that? Is it a real case?

Reply:
I do not understand the problem.
If several invoices come, we can:
  - Add lines to any of them (first, last, etc.)
  - Raise an error
  -...
Merge the invoices generated by the sale module is beyond the scope of this module

Reply:
OK

Reply:
Now I understand your questions.
Normally it is only grouped by partner and currency, but it is not a problem to control it in case some other module modifies that break of sales grouping

Reply:
Let's continue this way if standard doesn't "ungroup" easily invoices.

Reply:
After 'child_of' change it is possible to give the case easily

Reply:
OK, what you prefer